app/src/main/java/tftpserver/db/Db.java

```diff
@@ -32,4 +32,5 @@ public interface Db {
      * @throws FileNotFoundException If requested file isn't found
      */
     List<byte[]> getFile(String filename) throws FileNotFoundException;
+    boolean isPresent(String filename); 
```

Comment:
This looks incorrect functionally. The interface `Db` must just support `getFile` and `setFile`. The functionality you are trying to add here is implicitly covered in the definition of `getFile`, since it's defined to throw a `FileNotFoundException`. Please use that instead of adding the new `isPresent` method.